plugins/utils/rest.nix

```diff
@@ -434,9 +381,29 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       }
     ];
 
+    warnings =
+      let
+        definedOpts = lib.intersectLists (builtins.attrNames cfg.settings) [
+          "client"
+          "env_file"
+          "env_pattern"
+          "env_edit_command"
+          "encode_url"
+          "skip_ssl_verification"
+          "logs"
+          "result"
+        ];
```

Comment:
If you wanted to do nested options to, you could try something like:

```nix
definedOpts = filter (opt: hasAttrByPath (toList opt) cfg.settings) [
  "skip_ssl_verification"
  "skip_ssl_verification"
  [
  "logs"
    "logs"
    "level"
  ]
  [
    "logs"
    "save"
  ]
  # ...
];
```

TBH that's probably slightly more efficient too, since we aren't looking at _all_ attrs in `settings` (even the unrelated ones). Rather we only focus on the old ones we care about.

You'd also have to use `concatStringsSep "." (toList opt)` in the warning text.

Reply:
I thought about doing nested.. but realized it didn't really matter too much. We're not doing a rename to snake case transformation. But, it might help them identify the usage easier?

Reply:
Up to you. I just spotted you'd pushed a commit with some nested opts and realized you'd have to filter based on `hasAttrByPath` instead of just getting `attrNames`.

Reply:
If it's low effort it's probably best to check for any usage for options we used to document?